### PR TITLE
samples: matter: lto samples added

### PR DESCRIPTION
**Pull request by HubertHugo**
- Added .lto samples configs
- Adjusted other samples so thereis .debug and .release in every yaml

### CODE REVIEW

File: applications/matter_bridge/sample.yaml
Comment (doublemis1) [inline applications/matter_bridge/sample.yaml:1]: Maybe we should think about creating a separate debug/lto/release configuration for Matter Bridge with BLE bridged device support

File: samples/matter/lock/sample.yaml
Comment (doublemis1) [inline samples/matter/lock/sample.yaml:1]: Maybe add lto configuration for Matter WiFi/Thread Switch example

**Review by kkasperczyk-no — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]
LGTM!

File: samples/matter/lock/sample.yaml [hunk 1]
```diff
@@ -45,6 +78,28 @@ tests:
       - nrf5340dk/nrf5340/cpuapp
     platform_allow: nrf5340dk/nrf5340/cpuapp
     tags: sysbuild
+  sample.matter.lock.thread_wifi_switched.release.smp_dfu:
```
Comment (Damian-Nordic) [inline samples/matter/lock/sample.yaml:81]: The switched variant has LTO enabled by default.

**Review by LuDuda — APPROVED** [review]

**Pull request merged**